### PR TITLE
FLA-1027 - Handle Validation Errors

### PR DESCRIPTION
**Pull request by BrianMaki**
# Description

This PR includes the following proposed change(s):

- FLA-1027
- https://justice.gov.bc.ca/jira/browse/FLA-1027

- Converted return Response Entity creation to throw new *.Exception where validation were being done.
- Reordered alphabetically enums in ErrorCode.java.
- Removed enums no longer being used in ErrorResponse.java.
- Reordered alphabetically methods in relation to ErrorCode/Exception in ExceptionControllerAdvisor.java.
- Reordered alphabetically methods in relation to ErrorCode/Exception in ExceptionControllerAdvisorTest.java.
- Formatting changes (added space between casting class object and get) in ExceptionControllerAdvisorTest.java.
- Removed string value in throw new InvalidRoleException and replaced it with a constant string value.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Ran both unit tests in project as well as integration tests locally.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GenerateUrlTest.java [hunk 1]
```diff
@@ -310,9 +300,8 @@ public void whenApplicationCodeNotPresentShouldReturnForbidden() {
         initialPackage.setCourt(court);
         generateUrlRequest.setFilingPackage(initialPackage);
 
-        ResponseEntity<GenerateUrlResponse> actual = sut.generateUrl(transactionId, UUID.randomUUID().toString().replace("-", ""), TestHelpers.CASE_1, generateUrlRequest);
-
-        Assertions.assertEquals(HttpStatus.FORBIDDEN, actual.getStatusCode());
+        MissingApplicationCodeException exception = Assertions.assertThrows(MissingApplicationCodeException.class, () -> sut.generateUrl(transactionId, UUID.randomUUID().toString().replace("-", ""), TestHelpers.CASE_1, generateUrlRequest));
```
[the remark this answers is not in the document]
Comment (BrianMaki) [reply src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GenerateUrlTest.java:303]: Hi Taylor, do you know exactly what Code Climate is complaining about here? https://codeclimate.com/github/bcgov/jag-file-submission/pull/1939#

Lines 263, 282 and 303 are different test cases in different methods.

Line 263: 
InvalidRoleException exception = Assertions.assertThrows(InvalidRoleException.class, () -> sut.generateUrl(UUID.randomUUID(), UUID.randomUUID().toString().replace("-", ""), TestHelpers.CASE_3, generateUrlRequest));

Line 282: 
InvalidRoleException exception = Assertions.assertThrows(InvalidRoleException.class, () -> sut.generateUrl(transactionId, UUID.randomUUID().toString().replace("-", ""), TestHelpers.CASE_1, generateUrlRequest));

Line 303: 
MissingApplicationCodeException exception = Assertions.assertThrows(MissingApplicationCodeException.class, () -> sut.generateUrl(transactionId, UUID.randomUUID().toString().replace("-", ""), TestHelpers.CASE_1, generateUrlRequest));

[the remark this answers is not in the document]
Comment (BrianMaki) [reply src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GenerateUrlTest.java:303]: This is a unit test class which contains methods that have scenarios that look the same.

Lines 263, 282 and 303 are different test cases in different methods.

[the remark this answers is not in the document]
Comment (BrianMaki) [reply src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GenerateUrlTest.java:303]: Issues identified in code climate is based on code in test classes. Code in this test class will have scenarios that look the same but are not.

Lines 263, 282 and 303 are different test cases in different methods.

**Review by TayGov — APPROVED** [review]
👍

**Review by TayGov — APPROVED** [review]

**Pull request merged**